extension-files/tools/zabbix_discovery_devices

```diff
@@ -243,18 +243,31 @@ for filename in glob.glob('/sys/block/*'):
         for smartarg in config["smartctl_subdevices"][device_name]:
             device = {}
             device['{#BLOCKDEVICE}'] = device_name + " " + smartarg
+            device['{#DEVNAME}'] = device_name + " " + smartarg
+            device['{#DEVTYPE}'] = "disk"
             devices_discovered['data'].append(device)
     elif args.software_raid:
         if raid_disks_level:
             device = {}
             device['{#BLOCKDEVICE}'] = device_name
+            device['{#DEVNAME}'] = device_name
+            device['{#DEVTYPE}'] = "disk"
             device['{#RAIDLEVEL}'] = raid_disks_level.upper()
             devices_discovered['data'].append(device)
     else:
         device = {}
         device['{#BLOCKDEVICE}'] = device_name
+        device['{#DEVNAME}'] = device_name
+        device['{#DEVTYPE}'] = "disk"
         devices_discovered['data'].append(device)
 
+for filename in glob.glob('/sys/block/*/*/partition'):
```

Comment:
It might be a good idea to structure the discovery of partitions in the Zabbix template in a separate discovery or in a separate dashboard by making it controllable via a switch.


This would probably have the following advantages:


- The user can decide whether to collect the data for the partitions or not (to limit the item volume)

- the dashboards are clearer

- If templates have been set up on the discovery user (so that no unexpected things happen) 


There are also other switches where the general integration of partitions is not wanted. (see: `./zabbix_discovery_devices --help`, e.g. `--only_rawdisk`)


It would probably be best to reverse the logic and specify as a list what kind of devices you want to have in the discovery.

Reply:
My goal was simply to restore the [original functionality](https://www.zabbix.com/documentation/current/en/manual/discovery/low_level_discovery/examples/devices) of `vfs.dev.discovery`, because currently `zabbix-agent-extensions` is breaking other extensions relying on this function.
If you want the discovery to be more specific, my suggestion would be to  remove the [Alias](https://github.com/zabbix-tooling/zabbix-agent-extensions/blob/main/extension-files/agent-config/zabbix_disk_discovery.conf#L4) and start using `custom.vfs.dev.discovery` in discovery and templates etc.

Reply:
I have set the alias to be able to use the missing functionalities of the discovery with other templates, which is why it makes sense to keep this alias. As you can see, I didn't expect anyone to need this at partition level.

Then I can take your PR as it is because it completely restores the original behavior.

I will then add a switch to disable or deactivate the discovery of partitions.